azurerm/resource_arm_autoscale_setting.go

```diff
@@ -0,0 +1,978 @@
+		Tags:             expandedTags,
+		AutoscaleSetting: &autoscaleSetting,
+	}
+
+	result, err := asClient.CreateOrUpdate(ctx, resourceGroupName, name, parameters)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*result.ID)
+
+	return resourceArmAutoscaleSettingRead(d, meta)
+}
+
+func resourceArmAutoscaleSettingRead(d *schema.ResourceData, meta interface{}) error {
+	asClient := meta.(*ArmClient).autoscaleSettingsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["autoscalesettings"]
+
+	if name == "" {
+		return fmt.Errorf("Cannot find resource name in Resource ID for Autoscale Setting")
+	}
+
+	result, err := asClient.Get(ctx, resGroup, name)
+	if err != nil {
+		if result.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Autoscale Setting %s: %+v", name, err)
+	}
+
+	if result.AutoscaleSetting == nil {
+		return fmt.Errorf("Unexpected result when reading Autoscale Setting %s: %#v", name, result)
+	}
+	autoscaleSetting := *result.AutoscaleSetting
+	d.Set("name", result.Name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("location", result.Location)
+	d.Set("enabled", autoscaleSetting.Enabled)
+	d.Set("target_resource_id", autoscaleSetting.TargetResourceURI)
+	flattenAndSetTags(d, result.Tags)
+
+	d.Set("profile", flattenAzureRmAutoscaleProfile(autoscaleSetting.Profiles))
+	d.Set("notification", flattenAzureRmAutoscaleNotification(autoscaleSetting.Notifications))
+
+	return nil
+}
+
+func resourceArmAutoscaleSettingDelete(d *schema.ResourceData, meta interface{}) error {
+	armClient := meta.(*ArmClient)
+	asClient := armClient.autoscaleSettingsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroupName := id.ResourceGroup
+	autoscaleSettingName := id.Path["autoscalesettings"]
+
+	_, err = asClient.Delete(ctx, resourceGroupName, autoscaleSettingName)
+	return err
+}
+
+func expandAzureRmAutoscaleProfile(d *schema.ResourceData) ([]insights.AutoscaleProfile, error) {
+	profileData := d.Get("profile").([]interface{})
+	profiles := make([]insights.AutoscaleProfile, 0, len(profileData))
+
+	for _, profileConfig := range profileData {
+		profile := profileConfig.(map[string]interface{})
+
+		profileName := profile["name"].(string)
+		capacity := expandAzureRmAutoscaleCapacity(profile)
+		rules := expandAzureRmAutoscaleRule(profile)
+		autoscaleProfile := insights.AutoscaleProfile{
+			Name:     &profileName,
+			Capacity: &capacity,
+			Rules:    &rules,
+		}
+
+		fixedDate, fixedDateErr := expandAzureRmAutoscaleFixedDate(profile)
+		recurrence, recurrenceErr := expandAzureRmAutoscaleRecurrence(profile)
+
+		if fixedDate != nil && recurrence != nil {
+			return nil, fmt.Errorf("Conflict between fixed_date and reucrrence in profile %s", profileName)
```

Comment:
This logic could be simplified to the schema definition. Take [storage blob](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_storage_blob.go#L60) as an example.